tests/Aspire.Hosting.Tests/Cosmos/CosmosFunctionalTests.cs

```diff
@@ -0,0 +1,40 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.Tests.Helpers;
+using Polly;
+using Polly.Retry;
+using Xunit;
+
+namespace Aspire.Hosting.Tests.Cosmos;
+
+[Collection("IntegrationServices")]
+public class CosmosFunctionalTests
+{
+    private readonly IntegrationServicesFixture _integrationServicesFixture;
+
+    public CosmosFunctionalTests(IntegrationServicesFixture integrationServicesFixture)
+    {
+        _integrationServicesFixture = integrationServicesFixture;
+    }
+
+    [LocalOnlyFact()]
+    public async Task VerifyCosmosWorks()
+    {
+        var testProgram = _integrationServicesFixture.TestProgram;
+        var client = _integrationServicesFixture.HttpClient;
+
+        using var cts = new CancellationTokenSource(TimeSpan.FromMinutes(6));
+
+        await RetryPolicy.Handle<HttpRequestException>()
+                         .WaitAndRetryAsync(20, (count) => TimeSpan.FromSeconds(15))
```

Comment:
Can we move the retry logic into the IntegrationServiceA instead? That's where I am putting it in https://github.com/dotnet/aspire/pull/1540. IMO, it makes more sense there because if I was to run the app manually and hit the end-point it would still retry.

Also, waiting 15 seconds per retry is a bit long. That's like 5 minutes of retries. How long does the emulator take to start?

Reply:
Then you have to deal with the timeout logic in the httpclient from the fixture. The other reason for this is test debugability. It is easier to debug on the test side because you don't have to go an manually attach the debugger to the integrationservicea process to see what is going on.